sqlalchemy-stubs/sql/visitors.pyi

```diff
@@ -1,92 +1,110 @@
 from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Iterable
+from typing import Iterator
+from typing import List
+from typing import Optional
+from typing import TypeVar
+
+_ET = TypeVar("_ET", bound=ExternalTraversal)
+_T = TypeVar("_T")
 
 class TraversibleType(type):
     def __init__(cls, clsname: Any, bases: Any, clsdict: Any) -> None: ...
 
-class Traversible: ...
+class Traversible(metaclass=TraversibleType): ...
 
 class _InternalTraversalType(type):
     def __init__(cls, clsname: Any, bases: Any, clsdict: Any) -> None: ...
 
-class InternalTraversal:
-    def dispatch(self, visit_symbol: Any): ...
+class InternalTraversal(object, metaclass=_InternalTraversalType):
+    def dispatch(self, visit_symbol: Any) -> Optional[Callable[..., Any]]: ...
     def run_generated_dispatch(
         self,
         target: Any,
         internal_dispatch: Any,
         generate_dispatcher_name: Any,
-    ): ...
+    ) -> Any: ...
     def generate_dispatch(
         self,
         target_cls: Any,
         internal_dispatch: Any,
         generate_dispatcher_name: Any,
-    ): ...
-    dp_has_cache_key: Any = ...
-    dp_has_cache_key_list: Any = ...
-    dp_clauseelement: Any = ...
-    dp_fromclause_canonical_column_collection: Any = ...
-    dp_clauseelement_tuples: Any = ...
-    dp_clauseelement_list: Any = ...
-    dp_clauseelement_tuple: Any = ...
-    dp_executable_options: Any = ...
-    dp_fromclause_ordered_set: Any = ...
-    dp_string: Any = ...
-    dp_string_list: Any = ...
-    dp_anon_name: Any = ...
-    dp_boolean: Any = ...
-    dp_operator: Any = ...
-    dp_type: Any = ...
-    dp_plain_dict: Any = ...
-    dp_dialect_options: Any = ...
-    dp_string_clauseelement_dict: Any = ...
-    dp_string_multi_dict: Any = ...
-    dp_annotations_key: Any = ...
-    dp_plain_obj: Any = ...
-    dp_named_ddl_element: Any = ...
-    dp_prefix_sequence: Any = ...
-    dp_table_hint_list: Any = ...
-    dp_setup_join_tuple: Any = ...
-    dp_statement_hint_list: Any = ...
-    dp_unknown_structure: Any = ...
-    dp_dml_ordered_values: Any = ...
-    dp_dml_values: Any = ...
-    dp_dml_multi_values: Any = ...
-    dp_propagate_attrs: Any = ...
+    ) -> Any: ...
+    dp_has_cache_key: int = ...
+    dp_has_cache_key_list: int = ...
```

Comment:
technically these are an int subclass called _symbol(int), and these are interned on their value.   int is fine for now though...